Yoast/wordpress-seo
# Added Snippet editor to modal, added item in sidebar for access

## Summary

This PR can be summarized in the following changelog entry:

* Implements the Snippet Preview Editor in the sidebar, accessible through a modal. 

## Relevant technical choices:

* Added an extra component 'ButtonSection.js' (in Yoast Components) to the sidebar, based on the standard collapsible: showing a header and an icon. Clicking the component should open the modal. I say should because this part does not function correctly yet. I've added a `<p>` below the element to show the modal and onClick event work fine. Just for some reason not on the component. 

## Test instructions

This PR can be tested by following these steps:

* Check out this branch (same name) both in Yoast components and in Wordpress-seo. Run `yarn link` in the yoast components folder.
* In your wordpress-seo folder run `yarn link yoast-components` and build all the files. Run `yarn start` if you're running a dev envirionment. 
* Edit a post, on the right in the sidebar you should see a new component: "Snippet Preview". Clicking this should open a modal. This does not work so click the text "Snippet Preview Editor" below it. This should open a modal with the full Snippet Preview editor in it. The difference with the one in the metabox should be that it's always expanded and has no option to collapse it. It has two buttons to close it, both should work. 

## Quality assurance


Fixes #[707](https://github.com/Yoast/yoast-components/issues/707)
Requires #[712](https://github.com/Yoast/yoast-components/pull/712)
